diff --git a/dwm.c b/dwm.c
--- a/dwm.c
+++ b/dwm.c
@@ -1115,7 +1115,8 @@ monocle(Monitor *m)
 	if (n > 0) /* override layout symbol */
 		snprintf(m->ltsymbol, sizeof m->ltsymbol, "[%d]", n);
 	for (c = nexttiled(m->clients); c; c = nexttiled(c->next))
-		resize(c, m->wx, m->wy, m->ww - 2 * c->bw, m->wh - 2 * c->bw, 0);
+		resize(c, m->wx + gappx, m->wy + gappx,
+			m->ww - 2 * c->bw - 2 * gappx, m->wh - 2 * c->bw - 2 * gappx, 0);
 }
 
 void
@@ -1687,26 +1688,27 @@ tile(Monitor *m)
 {
 	unsigned int i, n, h, mw, my, ty;
+	unsigned int masters;
 	Client *c;
 
 	for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
 	if (n == 0)
 		return;
 
+	masters = MIN(n, m->nmaster);
 	if (n > m->nmaster)
-		mw = m->nmaster ? m->ww * m->mfact : 0;
+		mw = m->nmaster ? (m->ww - 2 * gappx - gappx) * m->mfact : 0;
 	else
-		mw = m->ww;
+		mw = m->ww - 2 * gappx;
 	for (i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
 		if (i < m->nmaster) {
-			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
-			resize(c, m->wx, m->wy + my, mw - (2*c->bw), h - (2*c->bw), 0);
-			if (my + HEIGHT(c) < m->wh)
-				my += HEIGHT(c);
+			h = (m->wh - 2 * gappx - my - (masters - i - 1) * gappx) / (masters - i);
+			resize(c, m->wx + gappx, m->wy + gappx + my, mw - 2 * c->bw, h - 2 * c->bw, 0);
+			my += HEIGHT(c) + gappx;
		} else {
-			h = (m->wh - ty) / (n - i);
-			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - (2*c->bw), h - (2*c->bw), 0);
-			if (ty + HEIGHT(c) < m->wh)
-				ty += HEIGHT(c);
+			h = (m->wh - 2 * gappx - ty - (n - i - 1) * gappx) / (n - i);
+			resize(c, m->wx + mw + 2 * gappx, m->wy + gappx + ty,
+				m->ww - mw - 3 * gappx - 2 * c->bw, h - 2 * c->bw, 0);
+			ty += HEIGHT(c) + gappx;
 		}
 }
